lib/gitarro/backend.rb

```diff
@@ -63,105 +63,119 @@ def initialize(option = nil)
-    return false unless changelog_changed(@repo, pr, comm_st)
+    return false unless changelog_changed(pr, comm_st)
     true
   end
 
   def unreviewed_pr_test(pr, comm_st)
     return unless unreviewed_pr_ck(comm_st)
-    pr_all_files_type(@repo, pr.number, @file_type)
+    pr_all_files_type(pr.number, @file_type)
     return if empty_files_changed_by_pr
     # gb.check is true when there is a job running as scheduler
     # which doesn't execute the test but trigger another job
     return false if @check
-    launch_test_and_setup_status(@repo, pr)
-    true
+    launch_test_and_setup_status(pr)
   end
 
   def reviewed_pr_test(comm_st, pr)
     # if PR status is not on pending and the context is not set,
     #  we dont run the tests
     return false unless context_pr(comm_st) == false ||
                         pending_pr(comm_st) == true
-    pr_all_files_type(@repo, pr.number, @file_type)
     return true if changelog_active(pr, comm_st)
-    return false unless @pr_files.any?
-    exit 1 if @check
-    launch_test_and_setup_status(@repo, pr)
-    true
-  end
-
-  private
-
-  # this function setup first pending to PR, then execute the tests
-  # then set the status according to the results of script executed.
-  # pr_head = is the PR branch
-  # base = is a the upstream branch, where the pr targets
-  def launch_test_and_setup_status(repo, pr)
-    # pending
-    @client.create_status(repo, pr.head.sha, 'pending',
-                          context: @context, description: @description,
-                          target_url: @target_url)
-    # do tests
-    @j_status = gbexec.pr_test(pr)
-    # set status
-    @client.create_status(repo, pr.head.sha, @j_status,
-                          context: @context, description: @description,
-                          target_url: @target_url)
+    return false unless pr_all_files_type(pr.number, @file_type).any?
+    @check ? exit(1) : launch_test_and_setup_status(pr)
   end
 
   # this function will check if the PR contains in comment the magic word
   # # for retrigger all the tests.
-  def magicword(repo, pr_number, context)
+  def magicword(pr_number, context)
     magic_word_trigger = "gitarro rerun #{context} !!!"
-    pr_comment = @client.issue_comments(repo, pr_number)
     # a pr contain always a comments, cannot be nil
-    pr_comment.each do |com|
+    @client.issue_comments(@repo, pr_number).each do |com|
       # delete comment otherwise it will be retrigger infinetely
       if com.body.include? magic_word_trigger
-        @client.delete_comment(repo, com.id)
+        @client.delete_comment(@repo, com.id)
         return true
       end
     end
     false
   end
 
+  private
+
+  # Create a status for a PR
+  def create_status(pr, status)
+    client.create_status(@repo, pr.head.sha, status, context: @context,
+                                                     description: @description,
+                                                     target_url: @target_url)
+    true
```

Comment:
i think this true should go on the respectives public method

`retrigger_pr_number` etc

Reply:
I think I placed this at the wrong place (create_status)

Shouldn't we place it a launch_test_and_setup_status since it's in use by all the public methods? In the end it's being used by all of them (with the exception of changelog) that are being used at /gitarro.rb